### PR TITLE
PMM-11912 DBaaS templates

### PR DESCRIPTION
**Pull request by rasika-chivate**
Add the missing information in the DBaaS templates doc:

https://jira.percona.com/browse/PMM-11912

### CODE REVIEW

**Review by gen1us2k — APPROVED** [review]

**Review by solovevayaroslavna — APPROVED** [review]

File: docs/dbaas/DBaaS_tepmplate.md [hunk 1]
```diff
@@ -1,3 +1,139 @@
...
+Examples
+
+- **Data clusters**: Different load patterns, such as simple reads, heavy writes, 50%/50% read/write, and the number of connections.
+
+- **Infrastructure** - Different parameters and tuning for the resulting cluster: such as network configuration (load balancing, exposure), storage classes/types, etc.
+
+- **Environment**: Combination of database cluster and infrastructure that will affect the database cluster configuration.
+
+
+# Customize PXC DB configuration
```
Comment (recharte) [inline docs/dbaas/DBaaS_tepmplate.md:14]: We should state that this is only an example

File: docs/dbaas/DBaaS_tepmplate.md [hunk 2]
```diff
@@ -1,3 +1,139 @@
...
+  - get
+  - list
+EOF
+
+Run the following command:
+
+```sh
+kubectl apply -f dbaas-operator-manager-role.yaml
+```
+
```
Comment (recharte) [inline docs/dbaas/DBaaS_tepmplate.md:106]: We are missing the Template CR creation step, see https://github.com/percona/dbaas-operator/blob/main/docs/templates.md#creating-the-pxctemplatepxcconfiguration-template-cr

File: docs/dbaas/DBaaS_tepmplate.md [hunk 3]
```diff
@@ -1,3 +1,139 @@
...
+DatabaseCluster Template allows operators to customize Database Clusters based on their requirements, environments, or infrastructure.
+
+Examples
+
+- **Data clusters**: Different load patterns, such as simple reads, heavy writes, 50%/50% read/write, and the number of connections.
+
+- **Infrastructure** - Different parameters and tuning for the resulting cluster: such as network configuration (load balancing, exposure), storage classes/types, etc.
+
+- **Environment**: Combination of database cluster and infrastructure that will affect the database cluster configuration.
```
Comment (recharte) [inline docs/dbaas/DBaaS_tepmplate.md:11]: This seems to duplicate what we had previously
> # Create a database cluster from a template
> Database clusters can be created from templates using PMM. Templates allow you to customize the creation of database clusters. You can adapt templates to tweak K8s-specific settings such as **liveness probes**, changing **config maps**, or tuning **database engines**.

File: docs/dbaas/DBaaS_tepmplate.md
Comment (recharte) [inline docs/dbaas/DBaaS_tepmplate.md:1]: Please rename the file to `DBaaS_template.md` otherwise the URL shows the typo.

**Review by recharte — CHANGES_REQUESTED** [review]
Thank you @rasika-chivate for updating the docs with the new example however I feel that we should follow a different approach.

You added an entirely new section for this but I feel that we should instead use the previous document structure and replace the old PXCTemplateUpgradeOptions example with the newer PXCTemplatePXCConfiguration example.

As you can see in [here](https://github.com/percona/dbaas-operator/blob/main/docs/templates.md#customizing-the-pxc-db-configuration) and [here](https://github.com/percona/dbaas-operator/blob/main/docs/templates.md#enabling-percona-xtradb-cluster-automatic-upgrade) both examples follow the same structure (1. create CRD; 2. add permissions; 3. create template CR) so replacing the old with the new one should be a drop-in replacement without much fuss.

File: docs/dbaas/DBaaS_tepmplate.md (hunk 1 above)
Comment (rasika-chivate) [reply docs/dbaas/DBaaS_tepmplate.md:14]: Done

File: docs/dbaas/DBaaS_tepmplate.md (hunk 3 above)
Comment (rasika-chivate) [reply docs/dbaas/DBaaS_tepmplate.md]: Done.

File: docs/dbaas/DBaaS_tepmplate.md [hunk 4]
```diff
@@ -1,85 +1,105 @@
...
 
-For the dbaas-operator to apply the template it needs access to the template CRs.
+In order for the dbaas-operator to apply the template it needs access to the template CRs:
 
 ```sh
-$ DBAAS_OPERATOR_MANAGER_ROLE=$(kubectl get clusterroles | grep dbaas-operator | grep -v metrics | grep -v proxy | cut -f 1 -d ' '); kubectl get clusterroles/"$DBAAS_OPERATOR_MANAGER_ROLE" -o yaml > dbaas-operator-manager-role.yaml
-
-$ cat <<EOF >>dbaas-operator-manager-role.yaml
+DBAAS_OPERATOR_MANAGER_ROLE=$(kubectl get clusterroles | grep dbaas-operator | grep -v metrics | grep -v proxy | cut -f 1 -d ' '); kubectl get clusterroles/"$DBAAS_OPERATOR_MANAGER_ROLE" -o yaml > dbaas-operator-manager-role.yaml
+cat <<EOF >>dbaas-operator-manager-role.yaml
```
Comment (recharte) [inline docs/dbaas/DBaaS_tepmplate.md:102]: Again, using the old example. Should be:

```th
DBAAS_OPERATOR_MANAGER_ROLE=$(kubectl get clusterroles | grep dbaas-operator | grep -v metrics | grep -v proxy | cut -f 1 -d ' '); kubectl get clusterroles/"$DBAAS_OPERATOR_MANAGER_ROLE" -o yaml > dbaas-operator-manager-role.yaml
cat <<EOF >>dbaas-operator-manager-role.yaml
- apiGroups:
  - dbaas.percona.com
  resources:
  - pxctemplatepxcconfiguration
  verbs:
  - get
  - list
EOF
```

Run the following command:

```sh
kubectl apply -f dbaas-operator-manager-role.yaml
```

**Review by recharte — CHANGES_REQUESTED** [review]

File: docs/dbaas/DBaaS_tepmplate.md [hunk 5]
```diff
@@ -1,127 +1,138 @@
...
+
+## Create Custom Resource Definition (CRD) template
+
+To create a template, do the following:
+
+1. Identify the field(s) of interest by reading the [PXC operator documentation](https://docs.percona.com/percona-operator-for-mysql/pxc/index.html) and the [PXC CRD](https://github.com/percona/percona-xtradb-cluster-operator/blob/v1.11.0/deploy/crd.yaml#L6378-L6379). In this case, you have to change the `spec.pxc.configuration` field.
+
+2. Create a template CRD `pxctpl-crd-pxc-configuration.yaml` with just those small subset of fields.
+
+        ```sh
```
Comment (recharte) [inline docs/dbaas/DBaaS_tepmplate.md:36]: This is rendered incorrectly, maybe is the tabs :shrug:

**Review by recharte — APPROVED** [review]

**Review by recharte — COMMENTED** [review]
Good job @rasika-chivate thank you

**Pull request merged**